subprojects/base-services/src/main/java/org/gradle/internal/resources/ProjectLeaseRegistry.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.resources;
+
+public interface ProjectLeaseRegistry {
+    /**
+     * Get a lock for the specified project.
+     *
+     * @param gradlePath
+     * @param projectPath
+     * @return the requested {@link ResourceLock}
+     */
+    ResourceLock getProjectLock(String gradlePath, String projectPath);
```

Comment:
Per our conversation, this is to avoid pulling this service down into core, which would be required if we made this method just take `Project` or some other type only available in core.